src/core/meshcop/meshcop.hpp

```diff
@@ -55,6 +57,155 @@ enum
 };
 
+/**
+ * This type represents Steering Data (bloom filter).
+ *
+ */
+class SteeringData : public otSteeringData
+{
+public:
+    enum
+    {
+        kMaxLength = OT_STEERING_DATA_MAX_LENGTH, ///< Maximum Steering Data length (in bytes).
+    };
+
+    /**
+     * This structure represents the hash bit index values for the bloom filter calculated from a Joiner ID.
+     *
+     * The first hash bit index is derived using CRC16-CCITT and second one using CRC16-ANSI.
+     *
+     */
+    struct HashBitIndexes
+    {
+        enum
+        {
+            kNumIndexes = 2, ///< Number of hash bit indexes.
+        };
+
+        uint16_t mIndex[kNumIndexes]; ///< The hash bit index array.
+    };
+
+    /**
+     * This method initializes the Steering Data and clears the bloom filter.
+     *
+     * @param[in]  aLength   The Steering Data length (in bytes) - MUST be smaller than or equal to `kMaxLength`.
+     *
+     */
+    void Init(uint8_t aLength = kMaxLength);
+
+    /**
+     * This method clears the bloom filter (all bits are cleared and no Joiner Id is accepted)..
+     *
+     * The Steering Data length (bloom filter length) is set to one byte with all bits cleared.
+     *
+     */
+    void Clear(void) { Init(1); }
+
+    /**
+     * This method sets the bloom filter to accept all Joiner IDs.
+     *
+     * To accept all Joiner IDs, The Steering Data length (bloom filter length) is set to one byte with all bits set.
+     *
+     */
+    void SetToAcceptAll(void);
+
+    /**
+     * This method returns the Steering Data length (in bytes).
+     *
+     * @returns The Steering Data length (in bytes).
+     *
+     */
+    uint8_t GetLength(void) const { return mLength; }
+
+    /**
+     * This method gets the Steering Data buffer (bloom filter).
+     *
+     * @returns A pointer to the Steering Data buffer.
+     *
+     */
+    const uint8_t *GetData(void) const { return m8; }
+
+    /**
+     * This method gets the Steering Data buffer (bloom filter).
+     *
+     * @returns A pointer to the Steering Data buffer.
+     *
+     */
+    uint8_t *GetData(void) { return m8; }
+
+    /**
+     * This method updates the bloom filter adding the given Joiner ID.
+     *
+     * @param[in]  aJoinerId  The Joiner ID to add to bloom filter.
+     *
+     */
+    void UpdateBloomFilter(const Mac::ExtAddress &aJoinerId);
+
+    /**
+     * This method indicates whether the bloom filter is empty (all the bits are cleared).
+     *
+     * @returns TRUE if the bloom filter is empty, FALSE otherwise.
+     *
+     */
+    bool IsEmpty(void) const { return DoesAllMatch(0); }
+
+    /**
+     * This method indicates whether the bloom filter accepts all Joiner IDs (all the bits are set).
+     *
+     * @returns TRUE if the bloom filter accepts all Joiners IDs, FALSE otherwise.
+     *
+     */
+    bool AcceptsAll(void) const { return (mLength > 0) && DoesAllMatch(kAcceptAll); }
```

Comment:
nit: since a bloom filter probabilistically indicates members of a set, how about using "Contains" instead of "Accepts" (here and everywhere).

Reply:
Thanks. SGTM. Changed in new pushed commit.